tests/foreman/ui/test_host.py

```diff
@@ -129,13 +130,18 @@ def setUpClass(cls):
         cls.proxy.organization = [cls.org_]
         cls.proxy = cls.proxy.update(['organization'])
 
-        # Search for domain and associate org, location
+        # Search for existing domain or create new otherwise. Associate org,
+        # location and dns to it
         _, _, domain = settings.server.hostname.partition('.')
         cls.domain = entities.Domain().search(
             query={
                 u'search': u'name="{0}"'.format(domain)
             }
-        )[0]
+        )
+        if len(cls.domain) == 1:
+            cls.domain = cls.domain[0]
+        else:
+            cls.domain = entities.Domain().create()
```

Comment:
Heh, clever way to introduce less changes to the code :) But still, this way to receive required domain you firstly create dummy and then update it with what's needed. How about doing that in one step? Like this:
```python
if len(cls.domain) == 1:
    cls.domain = cls.domain[0]
    cls.domain.location = [cls.loc]
    cls.domain.organization = [cls.org_]
    cls.domain.dns = cls.proxy
    cls.domain = cls.domain.update(['dns', 'location', 'organization'])
else:
    cls.domain = entities.Domain(
        dns=cls.proxy,
        location=[cls.loc],
        organization=[cls.org],
    ).create()
cls.domain_name = cls.domain.name
```